base/applications/rapps/available.cpp

```diff
@@ -301,8 +301,42 @@ BOOL CAvailableApps::UpdateAppsDB()
         return TRUE;
     }
 
-    DownloadApplicationsDB(APPLICATION_DATABASE_URL);
+    WCHAR * ApplicationDataBaseURL;
+    DWORD AppDataBaseURLLength = 0;
+    
+    if(SettingsInfo.bUseSource)
+    {
+        UrlCombineW(SettingsInfo.szSourceBaseURL,
+        APPLICATION_DATABASE_FILENAME,
+        0, &AppDataBaseURLLength, 0);
+        
+        ApplicationDataBaseURL = (WCHAR *)HeapAlloc(GetProcessHeap(), 0, AppDataBaseURLLength);
+        
+        if(!ApplicationDataBaseURL) return FALSE;
+        
+        UrlCombineW(SettingsInfo.szSourceBaseURL,
+        APPLICATION_DATABASE_FILENAME,
+        ApplicationDataBaseURL, &AppDataBaseURLLength, 0);
+    }
+    else
+    {
+        UrlCombineW(APPLICATION_DATABASE_BASEURL,
+        APPLICATION_DATABASE_FILENAME,
+        0, &AppDataBaseURLLength, 0);
+        
+        ApplicationDataBaseURL = (WCHAR *)HeapAlloc(GetProcessHeap(), 0, AppDataBaseURLLength);
+        
+        if(!ApplicationDataBaseURL) return FALSE;
+        
+        UrlCombineW(APPLICATION_DATABASE_BASEURL,
+        APPLICATION_DATABASE_FILENAME,
+        ApplicationDataBaseURL, &AppDataBaseURLLength, 0);
+    }
```

Comment:
I think it would be better if people have to specify the full url.
That way they can also specify a script or whatever type of web application that dynamically generates a file.

Reply:
I was thinking about probably some other files (for example, a file standalone to record dependencies between software, or an update for rapps itself) probably will be added in the future.
You are right, but I think people can still specify a web application to dynamically generate for ```rappmgr.cab``` if they want, just a little bit of trouble.

that's just my opinion, feel free to tell me if you still think full URL is better.😉

Reply:
Yes, for now there is one 'cab' file to be downloaded.
If we need a seperate file for dependencies (which I don't think), it can also be stored in this cab file.